File: MAP/cumparaturi/src/main/java/Service/ProdusService.java
package Service;

import Domain.Produs;
import Exceptions.RepositoryException;
import Repository.AbstractRepository;

import java.util.ArrayList;

public class ProdusService {
    private final AbstractRepository<Produs> produsRepo;

    public ProdusService(AbstractRepository<Produs> radioRepo) {
        this.produsRepo = radioRepo;
    }

    public void addProdus(int id, String marca, String nume, int pret, int cantitate) throws RepositoryException {

        if (produsRepo.findById(id))
            throw new RepositoryException("Exista deja un Produs cu id-ul = " + id);

        Produs newProdus = new Produs(id, marca, nume, pret, cantitate);

        produsRepo.add(newProdus);
    }

    public ArrayList<Produs> getProduse() throws NullPointerException{
        ArrayList<Produs> produse;
        produse = produsRepo.getEntities();
        if (produse.isEmpty()) {
            throw new NullPointerException( "There are no products yet!");
        }
        return produse;
    }

    public void deleteProdus(int id) throws RepositoryException {
        produsRepo.removeById(id);
    }

    public void updateProdus(int id, String marca, String nume, int pret, int cantitate) throws RepositoryException {

        Produs existingProdus = produsRepo.getEntityById(id);
        if (existingProdus == null) {
            throw new RepositoryException("Produs with id = " + id + " does not exist!");
        }

        Produs updatedProdus = new Produs(id, marca, nume, pret, cantitate);
        produsRepo.update(updatedProdus);
    }

    public Produs getProdusById(int id) {

        return produsRepo.getEntityById(id);
    }
}

File: FP/magazin/service/produs_service.py
from domain.entities import Produs


class ProdusService:
    def __init__(self, produs_file_repo, produs_validator):
        self.__produs_file_repo = produs_file_repo
        self.__produs_validator = produs_validator

    def show_products(self):
       return self.__produs_file_repo.show_products()

    def save(self, id, nume, pret):

        produs = Produs(id, nume, pret)
        self.__produs_validator.validate(produs)
        self.__produs_file_repo.save(produs)

    def delete_produs(self, id):
        self.__produs_file_repo.delete_produs(id)

    def update_produs(self, id, nume_nou, pret_nou):
        produs_nou = Produs(id, nume_nou, pret_nou)
        self.__produs_validator.validate(produs_nou)
        self.__produs_file_repo.update_produs(produs_nou)
